### PR TITLE
Fixed HyperVHypervisor.restore_vm()

### PR DESCRIPTION
**Pull request by Wiezzel**
Restoring VM would fail in case of insufficient RAM available. A memory check was added to shut down the VM in such situation (i.e. remove the saved state) and boot it normally.

### CODE REVIEW

File: tests/golem/docker/test_hyperv.py [hunk 1]
```diff
@@ -267,3 +268,67 @@ def test_check_smb_port_error(self, command, logger):
         self.hyperv._check_smb_port()
         command.assert_called_once()
         logger.error.assert_called_once()
+
+    def test_restore_vm_ok(self):
+        with patch.object(self.hyperv, '_check_memory') as check_memory, \
```
Comment (maaktweluit) [inline tests/golem/docker/test_hyperv.py:273]: i think the `@patch` decorator is neater here then `with patch`

**Review by maaktweluit — APPROVED** [review]
LGTM!

one nitty comment so still approved

File: tests/golem/docker/test_hyperv.py (hunk 1 above)
Comment (Wiezzel) [reply tests/golem/docker/test_hyperv.py:273]: How do I use `@patch` on `self.something`?

Comment (maaktweluit) [reply tests/golem/docker/test_hyperv.py:273]: good one, did not see that..

can only think of patching `HyperVHypervisor._check_memory`, but i am not sure if that works

**Review by mfranciszkiewicz — APPROVED** [review]

**Pull request merged**